msal/src/test/java/com/microsoft/identity/client/GlobalSettingsTest.java

```diff
@@ -0,0 +1,300 @@
+        Assert.assertEquals(AUTHORITY_URL_COMMON, authorities.get(0).getAuthorityURL().toString());
+        Assert.assertEquals(AUTHORITY_URL_TENANTED, authorities.get(1).getAuthorityURL().toString());
+    }
+    //endregion
+
+    // TODO: Once we figure out which fields should be global-only or pca-specific-only, add tests to make sure they can't be read in through other methods.
+
+    private GlobalSettings.GlobalSettingsListener getSuccessGlobalListener() {
+        return new GlobalSettings.GlobalSettingsListener() {
+            @Override
+            public void onSuccess(@NonNull String message) {
+                // Nothing
+            }
+
+            @Override
+            public void onError(@NonNull MsalException exception) {
+                Assert.fail(exception.getMessage());
+            }
+        };
+    }
+
+    private GlobalSettings.GlobalSettingsListener getSecondInitFailureGlobalListener() {
+        return new GlobalSettings.GlobalSettingsListener() {
+            @Override
+            public void onSuccess(@NonNull String message) {
+                Assert.fail("Second initialization was allowed.");
+            }
+
+            @Override
+            public void onError(@NonNull MsalException exception) {
+                Assert.assertEquals(exception.getErrorCode(), GlobalSettings.GLOBAL_ALREADY_INITIALIZED_ERROR_CODE);
+                Assert.assertEquals(exception.getMessage(), GlobalSettings.GLOBAL_ALREADY_INITIALIZED_ERROR_MESSAGE);
+            }
+        };
+    }
+
+    private GlobalSettings.GlobalSettingsListener getGlobalAfterPcaFailureGlobalListener() {
+        return new GlobalSettings.GlobalSettingsListener() {
+            @Override
+            public void onSuccess(@NonNull String message) {
+                Assert.fail("Second initialization was allowed.");
+            }
+
+            @Override
+            public void onError(@NonNull MsalException exception) {
+                Assert.assertEquals(exception.getErrorCode(), GlobalSettings.GLOBAL_INIT_AFTER_PCA_ERROR_CODE);
+                Assert.assertEquals(exception.getMessage(), GlobalSettings.GLOBAL_INIT_AFTER_PCA_ERROR_MESSAGE);
+            }
+        };
+    }
+
+    private void createSAPCAFromConfigPath(final String configPath, final CountDownLatch countDownLatch){
+        final File configFile = new File(configPath);
+        PublicClientApplication.createSingleAccountPublicClientApplication(mContext, configFile, new IPublicClientApplication.ISingleAccountApplicationCreatedListener() {
+            @Override
+            public void onCreated(ISingleAccountPublicClientApplication application) {
+                mSingleAccountApplication = application;
+                countDownLatch.countDown();
+            }
+            @Override
+            public void onError(MsalException exception) {
+                Assert.fail(exception.getMessage());
+            }
+        });
+
+        flushScheduler();
```

Comment:
still surprises me why we would need this as we start using countDownLatch.. @shahzaibj  any idea?

Reply:
Yes, im still confused why I needed to use flushScheluder() here, if I don't have it then the countDownLatch.await() hangs forever. I don't think I tried it with countDownLatch.await(TIMEOUT), though.

Reply:
I tried this again with the timeout, getting this error:

`Main looper has queued unexecuted runnables. This might be the cause of the test failure. You might need a shadowOf(getMainLooper()).idle() call.`

This was appearing in an earlier PR for me, the only solution that worked for me in that case was adding back the flushScheduler().

Reply:
In my experience, countdown latches don't work with Robolectric at all and for that reason I've never used them. `flushScheduler` should be the way to go.